distribution/tools/upgrade-cli/README.md

```diff
@@ -0,0 +1,38 @@
+OpenSearch Upgrade CLI
+===
+
+While upgrading to OpenSearch from a compatible version of Elasticsearch, there are some manual tasks that need to be performed.
+The `opensearch-upgrade` tool included in the distribution automates those tasks to make the upgrade process easier, consistent and error-free.
+It provides the following features,
+* imports the existing configurations and applies it to the new installation of OpenSearch.
+* installs the existing core plugins.
+
+However, **please note that**
+* The tool doesn't perform an end-to-end upgrade but rather assists during an upgrade process. Stopping of the existing service and starting the OpenSearch service must be done manually. It must be run on each node of the cluster individually as part of the upgrade process.
+* The tool doesn't provide a rollback option once you've upgraded a node. So make sure you follow best practices and take backups.
+* All community plugins must be installed (if available) manually.
+* The keystore settings can only be validated at service start up time, so any unsupported setting that is imported must be removed manually for the service to start.
+
+Usage
+---
+Make sure the following environment variables are available in the current environment
+
+* `ES_HOME` - path to existing Elasticsearch installation home
+* `ES_PATH_CONF` - path to the config directory of existing Elasticsearch installation
+* `OPENSEARCH_HOME` - path to the OpenSearch installation home
+* `OPENSEARCH_PATH_CONF` - path to the OpenSearch config directory
+
+The `opensearch-upgrade` is included in the `bin` directory. This must be run by the same user running the current Elasticsearch service. On each node of the cluster, the tool can be run by issuing the command.
+
+```
+$OPENSEARCH_HOME/bin/opensearch-upgrade
+```
+
+The tool sequentially performs the following tasks,
+
+1. Looks for a valid Elasticsearch installation on the current node. It reads the `elasticsearch.yml` to get the endpoint details to connect to the locally running Elasticsearch service. If it is unable to do so, it tries to retrieve the information from the `ES_HOME` location.
+2. Verifies if the existing version of Elasticsearch is compatible with the OpenSearch version. It prints a summary of the information gathered to the user and prompts for confirmation to proceed.
+3. Imports the settings from the `elasticsearch.yml` config file into the `opensearch.yml` config file.
+4. Copies across any custom JVM options from the `$ES_PATH_CONF/jvm.options.d` directory into the `$OPENSEARCH_PATH_CONF/jvm.options.d`. Similarly, It also imports the logging configurations from the `$ES_PATH_CONF/log4j2.properties` into `$OPENSEARCH_PATH_CONF/log4j2.properties`.
+5. Installs the core plugins that are currently installed in the `$ES_HOME/plugins` directory. It only installs the official OpenSearch plugins. *All other third party community plugins must be installed manually.*
```

Comment:
A few questions regarding the cli from reading this doc:
What if I was running elasticsearch-oss with no plugins? (no security plugin either) will the end result be an opensearch cluster with no plugins installed? can I choose to use the `opensearch-min` version?
What are the _official opensearch plugins_?   
OR
does this mean that if I was using the `repository-s3` plugin, for example, then at the end I would have opensearch with just that plugin installed?

Reply:
>> What if I was running elasticsearch-oss with no plugins? (no security plugin either) will the end result be an opensearch cluster with no plugins installed? 

Yes

>> can I choose to use the opensearch-min version?

Yes

>> What are the official opensearch plugins?

I mean the plugins that are part of the OpenSearch repository and can be installed by name using the `opensearch-plugin` tool. 

>> does this mean that if I was using the repository-s3 plugin, for example, then at the end I would have opensearch with just that plugin installed?

Yes, unless the new OpenSeach installation already plugins in it as the tool doesn't remove any plugins.

Reply:
@adnapibar Awesome! thanks!